### ISSUE TITLE
Basic programming competency not expressed by the main developer

### ISSUE THREAD

**Issue by phillips1012** [ShiftOS/Project-Orion#1]
I'm sorry for this petty issue title, but this has to be resolved if you ever plan for this to work out.

I was looking through the code and have come to the conclusion that this rewrite will be no less messy than the original.

- Example: https://github.com/ShiftOS/Project-Orion/blame/4f14653161e122ba88406bbae1a7bbb5f40f9e7c/ShiftOS/ShiftOS/Terminal.vb#L104-L138    
`Else If` should be used, but the fact that it is not shows that basic programming competency is missing.

- Example: https://github.com/ShiftOS/Project-Orion/blame/4f14653161e122ba88406bbae1a7bbb5f40f9e7c/ShiftOS/ShiftOS/Paths.vb#L28    
Hard-coded paths were one of the main issues of the original game. `C:\ShiftOS\` should merely be a suggestion to the user, and therefore should be view logic. A better place than `C:\ShiftOS\` would be `%LOCALAPPDATA%\ShiftOS`, anyways.

**Comment by TheUltimateHacker** [comment]
Thanks for the idea. By default, the root is C:\ShiftOS\ however in the future I'm going to allow users to change the root folder in-game during Setup, so this is temporary.

As for the Terminal, I have tried
If command = "this" Then
   'do this
ElseIf command = "that" Then
  'do that
End If
however this has backfired on me where entering "that" for example, would be taken in as an invalid command.

**Comment by pfgithub** [comment]
look up switch case in vb, as well as splitting a string at spaces

**Comment by TheUltimateHacker** [comment]
Fixed the issue, issue closed.

**Comment by phillips1012** [comment]
That particular messy code bit was not what the issue was about.

**Comment by V0rTeXbg** [comment]
@RandomMelon Have you seen phillips's comment?

**Comment by RandomMelon** [comment]
@V0rTeXbg I have, don't really know what he's talking about, as the main reason we're doing this is to fix the messy code issue in ShiftOS, so eventually all messy code will be erased in this new project.

**Comment by phillips1012** [comment]
The problem that you're clearly not seeing is that this is not a refactor of the original shiftos.

This is a full on rewrite, with messy code written directly from the main developer. You cannot start out with messy code then work from there to "de-messy" it, order has to be made early on or else the whole project will fail.

**Comment by UberGM** [comment]
stop closing the issue

**Comment by RandomMelon** [comment]
TheUltimateHacker said he fixed the issue, so stop reopening the issue, I think phillips got his point to us and we don't need more comments.

**Comment by pfgithub** [comment]
it's still not fixed

**Comment by phillips1012** [comment]
...Do you not know how issues work? You keep them open until they're resolved. Both issues I presented, are not fixed:

1. The messy code example I pointed out
2. The fact that developers don't seem capable of producing quality code

**Comment by V0rTeXbg** [comment]
@RandomMelon If you lock this issue one more time i will have to temporary remove your rights. Beware.

**Comment by RandomMelon** [comment]
Comments aren't really needed as of now, we're using #Regions to group code. Oh yeah, that's only in my commit.

**Comment by phillips1012** [comment]
Comments weren't used where appropriate. The confusing part of `Terminal.vb` was not commented to explain why a `further` variable was used..

Defining regions is not an alternative for actually explaining confusing logic using comments.

**Comment by RandomMelon** [comment]
Yep, I know. I haven't coded in VB.NET for ages, so yeah blame that fact.
I'll comment it now :)

**Comment by phillips1012** [comment]
What does commenting frequency have to do with what programming language you're using?

**Comment by RandomMelon** [comment]
Nothing.. lol haven't coded in ages I should've said sorry about that

**Comment by phillips1012** [comment]
Which goes back to programming competency...

**Comment by RandomMelon** [comment]
Indeed. So it does.. but I'm not the lead developer, TheUltimateHacker is.

**Comment by RandomMelon** [comment]
I'm finished commenting the Terminal. I'm gonna commit it now.

**Comment by phillips1012** [comment]
You forgot to push...

**Comment by RandomMelon** [comment]
I can't push.. I'm under some merge issues :(

**Comment by rezedus** [comment]
https://github.com/ShiftOS/Project-Orion/blob/bc77f1783f75d67177f4aec776634c30ca514f3e/ShiftOS/ShiftOS/Terminal.vb versus https://github.com/ShiftOS/Project-Orion/blob/master/ShiftOS/ShiftOS/Terminal.vb
https://github.com/ShiftOS/Project-Orion/blob/bc77f1783f75d67177f4aec776634c30ca514f3e/ShiftOS/ShiftOS/ShiftOS.vb versus https://github.com/ShiftOS/Project-Orion/blob/master/ShiftOS/ShiftOS/ShiftOS.vb

I demand an explanation why were my commits reverted.

**Comment by RandomMelon** [comment]
Merge issues :(

**Comment by rezedus** [comment]
> claims that "the main reason we're doing this is to fix the messy code issue in ShiftOS, so eventually all messy code will be erased in this new project"
> makes messy code

https://github.com/ShiftOS/Project-Orion/blob/c791843fdccb721fad7b435dde2b72f8af756ef2/ShiftOS/ShiftOS/Terminal.vb#L194-L237

Regarding the regions talk, I made issue #4 for that.

**Comment by NarodGaming** [comment]
I fixed that @rezedus
